### PR TITLE
fix(irs-api):[TRI-631] Added method to find and delete cancelled jobs

### PR DESCRIPTION
**Pull request by ds-ext-ooyeneye**

### CODE REVIEW

**Comment by ds-ext-kmassalski** [comment]
The bug was completely different, and has nothing to do with canceled jobs.
Please read one more time bug/task description.

**Review by ds-ext-kmassalski — CHANGES_REQUESTED** [review]
The bug was completely different, and has nothing to do with canceled jobs.
Please read one more time bug/task description.

File: irs-api/src/main/java/net/catenax/irs/connector/job/JobOrchestrator.java [hunk 1]
```diff
@@ -202,6 +204,20 @@ public void findAndCleanupFailedJobs() {
         log.info("Deleted {} failed jobs", multiTransferJobs.size());
     }
 
+    @IrsTimer("cleancancelledjobs")
+    @Scheduled(cron = "${irs.job.cleanup.scheduler.completed}")
+    public void findAndCleanupCancelledJobs() {
```
Comment (ds-ext-kmassalski) [inline irs-api/src/main/java/net/catenax/irs/connector/job/JobOrchestrator.java:209]: Currently there is no requirement that we should remove canceled jobs, additionaly it was not in scope of this task.
So please remove this scheduler method.

File: irs-api/src/main/java/net/catenax/irs/persistence/MinioBlobPersistence.java [hunk 2]
```diff
@@ -172,6 +173,16 @@ public boolean delete(final String sourceBlobName) throws BlobPersistenceExcepti
         }
     }
 
+    private void deleteRelatedJobsByProcessId(final List<String> processIds) throws BlobPersistenceException {
```
Comment (ds-ext-kmassalski) [inline irs-api/src/main/java/net/catenax/irs/persistence/MinioBlobPersistence.java:176]: deleteRelatedJobsByProcessId - is not the best name, as inside this function we are not 'deleting jobs', but rather items/processes.
Please change to more suitable:
deleteRelatedWithJobItems or deleteRelatedWithJobProcesses

File: irs-api/src/test/java/net/catenax/irs/persistence/MinioBlobPersistenceTest.java [hunk 3]
```diff
@@ -69,13 +72,26 @@ void shouldReturnFalseWhenDeletingNonexistentBlob() throws Exception {
...
         // assert
         verify(client).removeObject(any());
         assertThat(success).isFalse();
     }
 
+    @Test
+    void shouldDeleteBlobWithRelatedBlobByProcessId() throws Exception {
+        final List<String> processIds = List.of("testBlobName_process_1", "testBlobName_process_2");
+        testee.putBlob(processIds.get(0), "testContent".getBytes(StandardCharsets.UTF_8));
+        testee.putBlob(processIds.get(0), "testContent".getBytes(StandardCharsets.UTF_8));
```
Comment (ds-ext-kmassalski) [inline irs-api/src/test/java/net/catenax/irs/persistence/MinioBlobPersistenceTest.java:86]: I suppose you wanted: processIds.get(1) here

**Review by ds-ext-kmassalski — CHANGES_REQUESTED** [review]

File: irs-api/src/main/java/net/catenax/irs/connector/job/JobOrchestrator.java (hunk 1 above)
Comment (ds-ext-ooyeneye) [reply irs-api/src/main/java/net/catenax/irs/connector/job/JobOrchestrator.java:209]: But can be eventually needed

File: irs-api/src/main/java/net/catenax/irs/persistence/MinioBlobPersistence.java (hunk 2 above)
Comment (ds-ext-ooyeneye) [reply irs-api/src/main/java/net/catenax/irs/persistence/MinioBlobPersistence.java:176]: But you are still using Job within the method name, maybe a rather befitting name could be deleteRelatedBlobs

File: irs-api/src/test/java/net/catenax/irs/connector/job/PersistentJobStoreTest.java [hunk 4]
```diff
@@ -492,4 +492,21 @@ void checkLastModifiedOnAfterCreation() {
...
 
+    @Test
+    void deleteJobsAndRelatedDataInStorage() {
+        sut.create(job);
+
+        sut.addTransferProcess(job.getJobId().toString(), processId1);
+        sut.completeTransferProcess(job.getJobIdString(), process1);
+        sut.completeJob(job.getJobIdString(), this::doNothing);
+
+        final int sizeAfterCreation = blobStoreSpy.findBlobByPrefix("job:").size();
```
[the remark this answers is not in the document]
Comment (ds-ext-kmassalski) [reply irs-api/src/test/java/net/catenax/irs/connector/job/PersistentJobStoreTest.java:503]: Expected test casse:
- register job that will be completed
- count items in blob store - ~10 items
- delete/remove process
- count items in blob store - 0 items

**Review by ds-ext-kmassalski — CHANGES_REQUESTED** [review]

**Comment by ds-ext-ooyeneye** [comment]
An extended item search from the bucket was done after deleting the job item from the bucket, using the job id,  complete transfer id, and process id. Should no item be gotten from this search and eventually there are still items related to the job in the bucket, then we dealing with Item gone woe, (i.e. a wild Item), the bug is then beyond IRS.

File: irs-api/src/test/java/net/catenax/irs/connector/job/PersistentJobStoreTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (ds-ext-ooyeneye) [reply irs-api/src/test/java/net/catenax/irs/connector/job/PersistentJobStoreTest.java]: The above test was 0 but does not solve the problem of related item been found in the bucket

**Review by ds-jhartmann — APPROVED** [review]

**Review by ds-ext-kmassalski — APPROVED** [review]

**Pull request merged**